jumpscale/packages/vdc_dashboard/chats/matic.py

```diff
@@ -94,17 +103,43 @@ def _enter_sentry_info(self):
         self.config.chart_config.extra_config["sentry_nodeid"] = sentry_nodeid.value
         self.config.chart_config.extra_config["sentry_enodeid"] = sentry_enodeid.value
 
+    @chatflow_step(title="Web Access")
     def _enter_access_code(self):
         self.config.chart_config.access_code = self.secret_ask(
             "Enter Access Code (This would be used to access your node's web page)", min_length=8, required=True
         )
 
+    @chatflow_step(title="Port Settings")
+    def _get_node_ports(self):
+        usedports = get_ingresstcp_used_ports()
+        msg = "For multiple deployments, please ensure to use different ports. TCP Ports already in use are " + str(
+            usedports
+        )
+        form = self.new_form()
```

Comment:
i think this needs to go into a while loop that exits when the user chooses valid port, instead of starting the chatflow all over again

Reply:
Done with https://github.com/threefoldtech/js-sdk/pull/3082/commits/626fc327c70e9e0b600eefe609fd58a63b31dce6